### DISCUSSION TITLE
Should initFirstItem be deleted in production?

### DISCUSSION THREAD

**Discussion by adrianbienias** [keystonejs/keystone#8492] [Questions] [answered]
The default [`auth.ts`](https://github.com/keystonejs/create-keystone-app/blob/main/create-keystone-app/starter/auth.ts#L42) states:
`WARNING: remove initFirstItem functionality in production`

On the other hand, when the app is deployed in production for the first time, this mechanism is useful.

Should it really be removed in production, even after the first 'admin' user is created?

By that warning, I understand that the suggested workflow looks like the following:

1. Deploy the app in production
2. Create the first user (admin)
3. Update the code by removing `initFirstItem`

But I have doubts that the 3rd step is necessary.

**Comment by arsekil** [comment]
I suggest you create an admin user in devmode, then remove initFirstItem and keep your database intact, then deploy your database as well and it should work.

**Reply by adrianbienias** [reply]
Basically my question is - does `initFirstItem` is really some kind of security issue even if the first item is already created?

Apart from that, I don't get why you suggest creating an admin user in dev mode?

**Comment by dcousens** [comment, marked as answer]
The recommendation to remove `initFirstItem` is that the conditions for that function to be called might not coincide with the operating assumptions of your software system.

Typically you only want that functionality as a way to try out Keystone or quickly start a new project.
After that, as your software architecture changes, you might not uphold the conditions that are required for that function to remain dormant and that could be a security problem.

If you don't need it, remove it.

**Reply by adrianbienias** [reply]
That's a pity.

My use case is giving a client a package (e.g. docker image) that they can just run and start using.

`initFirstItem` is awesome for that case to give a client an easy way to create their first admin user.

But if that functionality has to be removed, it's not suitable for that case because a client don't want to manually update the code in their package.
